### PR TITLE
Actions sms

### PR DESCRIPTION
**Pull request by martinsmid**
Please do not merge yet, I will use this PR for comments and questions

### CODE REVIEW

File: actions-sms.php [hunk 1]
```diff
@@ -241,140 +233,135 @@ function freeBikes($number)
 
 function log_sms($sms_uuid, $sender, $receive_time, $sms_text, $ip)
 {
-    global $dbserver,$dbuser,$dbpassword,$dbname;
```
Comment (martinsmid) [inline actions-sms.php]: Is this ok ? I saw in other files that these credentials are used for the main db too, so I removed this db creation.

File: actions-sms.php [hunk 2]
```diff
@@ -241,140 +233,135 @@ function freeBikes($number)
...
-    $bikeNum=trim($bikeNum);
+    $bikeNum = trim($bikeNum);
     if (preg_match("/^[0-9]*$/", $bikeNum)) {
         $bikeNum = intval($bikeNum);
     } elseif (preg_match("/^[A-Z]+[0-9]*$/i", $bikeNum)) {
         $standName = $bikeNum;
         delstandnote($number, $standName, $message);
         return;
     } else {
-        sendSMS($number, _('Error in bike number / stand name specification:'.$db->conn->real_escape_string($bikeNum)));
```
Comment (martinsmid) [inline actions-sms.php]: I moved all escaping to the redbean getAll/exec statements, so I guess the db queries are ok. Did I miss something?

File: actions-sms.php [hunk 3]
```diff
@@ -432,226 +417,221 @@ function delstandnote($number, $standName, $message)
...
-
-    $result=R::getAll("SELECT userName from users where number=:number", [':number' => $number]);
-    $row =$result->fetch_assoc();
-    $reportedBy=$row["userName"];
+    $row = $result[0];
+    $standId = $row["standId"];
 
+    $result = R::getAll("SELECT userName from users where number=:number", [':number' => $number]);
+    $row = $result[0];
+    $reportedBy = $row["userName"];
```
Comment (martinsmid) [inline actions-sms.php]: Isn't this the same (user) as $user_id ? Where is the `getUser` defined ? Can the user be retrieved at once with id and name ?

File: actions-sms.php (hunk 1 above)
Comment (nekromoff) [inline actions-sms.php]: yes, this was used before only to connect to the master db, obsoleted by redbeanphp

File: actions-sms.php (hunk 2 above)
Comment (nekromoff) [inline actions-sms.php]: seems to be fine to me

File: actions-sms.php (hunk 3 above)
Comment (nekromoff) [inline actions-sms.php]: yes, this seems to be the same.
see file common.php and these functions that should be used all over:
`function getprivileges($userid)
{
    $user=R::load('users', $userid);
    if ($user->id) {
        return $user->privileges;
    }
    return false;
}

function getusername($userid)
{
    $user=R::load('users', $userid);
    if ($user->id) {
        return $user->username;
    }
    return false;
}

function getphonenumber($userid)
{
    $user=R::load('users', $userid);
    if ($user->id) {
        return $user->number;
    }
    return false;
}

function getuserid($number)
{
    $user=R::findOne('users', 'number=?', [$number]);
    if (!empty($user)) {
        return $user->id;
    }
    return false;
}`

**Pull request merged**